sphinx/texinputs/sphinxlatexadmonitions.sty

```diff
@@ -194,19 +193,18 @@
   {\begin{sphinxheavybox}\sphinxstrong{#1} }{\end{sphinxheavybox}}
 \newenvironment{sphinxerror}[1]
   {\begin{sphinxheavybox}\sphinxstrong{#1} }{\end{sphinxheavybox}}
-% or just use package options
 
 % the main dispatch for all types of notices
 \newenvironment{sphinxadmonition}[2]{% #1=type, #2=heading
   % can't use #1 directly in definition of end part
   \def\spx@noticetype {#1}%
-  % set parameters of heavybox/lightbox
+  % set the parameters common to lightbox and heavybox environments
   \sphinxcolorlet{spx@notice@bordercolor}{sphinx#1BorderColor}%
   \sphinxcolorlet{spx@notice@bgcolor}{sphinx#1BgColor}%
   \spx@notice@border \dimexpr\csname spx@#1@border\endcsname\relax
-  % start specific environment, passing the heading as argument
+  % trigger the sphinx<type> environment, #2=heading is passed as argument
   \begin{sphinx#1}{#2}}
-  % workaround some LaTeX "feature" of \end command
- {\edef\spx@temp{\noexpand\end{sphinx\spx@noticetype}}\spx@temp}
+  % workaround some LaTeX "feature" of \end command (can't use "sphinx#1" here)
+ {\expanded{\noexpand\end{sphinx\spx@noticetype}}}
```

Comment:
I should not have use `\expanded` here, as it is not available prior to 2019 (except with lualatex).